### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sumitpannalall**

### CODE REVIEW

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    public static LiSecuredPrefManager getInstance() {
+        if (_instance == null) {
+            throw new NoSuchPropertyException("LiSecuredPrefManager not intialized. Call init method first");
+        }
+        return _instance;
+    }
+
+    private LiSecuredPrefManager(Context context) throws NoSuchAlgorithmException {
+        encryptionKey = "$2a$12$LHZACI8yh9OZDK0Ep.gb9u".getBytes();
+//        String encryptionStr = context.getPackageName().toCharArray() + Settings.Secure.getString(context.getContentResolver(),
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java]: remove commented lines

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            this.getSecuredPreferences(context).edit().putString(key, value).apply();
+        }
+    }
+
+    public String getString(Context context, String key) {
+        //get secured preferences and check key has any value there
+        SharedPreferences securedPreferences = this.getSecuredPreferences(context);
+        String value;
+        try {
+            value = LiSecuredPrefManager.getInstance().decrypt(LiSecuredPrefManager.getInstance().encrypt(key));
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java]: Why key is encrypted and then decrypted in same line? This line is  intended to fetch value against any key, but i don't see any code which is fetching value for the given key from shared preference.

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/model/response/LiUser.java [hunk 3]
```diff
@@ -351,7 +353,9 @@ public JSONObject jsonSerialize() {
         LiCoreSDKUtils.put(json, USER_AVATAR, this.avatar.jsonSerialize());
         LiCoreSDKUtils.put(json, USER_LOGIN, this.login.getValue());
         LiCoreSDKUtils.put(json, USER_HREF, this.href);
-        LiCoreSDKUtils.put(json, USER_VIEW_HREF, this.profilePageUrl);
+        if (!TextUtils.isEmpty(profilePageUrl)) {
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/model/response/LiUser.java]: else is not requried ?

File: li-android-sdk-core/src/test/java/lithium/community/android/sdk/auth/LiAuthServiceTest.java [hunk 4]
```diff
@@ -267,26 +267,6 @@ public void aperformAuthorizationRequestSSOTest() throws Exception {
         doNothing().when(liAuthService,"dispose");
 
         LiAuthorizationException liAuthorizationException=null;
-        doAnswer(
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/test/java/lithium/community/android/sdk/auth/LiAuthServiceTest.java]: enablePostAuthorizationFlows is still in LiAuthService, will not this change affect the testcase ?

Comment (sumitpannalall) [inline li-android-sdk-core/src/test/java/lithium/community/android/sdk/auth/LiAuthServiceTest.java]: enablePostAuthorizationFlows just broadcasts an Intent which needs to be registered with other flows. This won't affect the functionality of LiAuthServiceImpl
`        Intent intent = new Intent(mContext.getString(R.string.li_login_complete_broadcast_intent));
        intent.putExtra(LiCoreSDKConstants.LOGIN_RESULT, isLoginSuccess);
        intent.putExtra(LiCoreSDKConstants.LOGIN_RESULT_CODE, responseCode);
        mContext.sendBroadcast(intent);
        this.dispose();
`

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/model/response/LiUser.java (hunk 3 above)
Comment (sumitpannalall) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/model/response/LiUser.java]: if profilePageUrl is not set then we don't serialize that.

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    private static final String ENCRYPTION_ALGORITHM = "AES";
+    private static final int ENCRYPTION_LENGTH = 16;
+    private byte[] encryptionKey;
+
+    /**
+     * Instance of this.
+     */
+    public static LiSecuredPrefManager getInstance() {
+        if (_instance == null) {
+            throw new NoSuchPropertyException("LiSecuredPrefManager not intialized. Call init method first");
```
Comment (sumitpannalall) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiSecuredPrefManager.java]: removed throwing of RunTimeException and now returning null if the initialization wasn't done properly

**Pull request merged**